main.cpp

```diff
@@ -18,13 +18,24 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private: Rewriter &rewriter;
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& rewriter_):rewriter(rewriter_) {
         // Your code goes here
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+        const auto *StyleCastExpr=Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        auto leftParenLoc=StyleCastExpr->getLParenLoc();
+        auto rigthParenLoc=StyleCastExpr->getRParenLoc();
+        rewriter.RemoveText(leftParenLoc,1);
+        rewriter.RemoveText(rigthParenLoc,1);
+        auto &sourceManager=*Result.SourceManager;
+        auto endLine=Lexer::getLocForEndOfToken(StyleCastExpr->getEndLoc(),0,sourceManager,LangOptions());

```

Comment:
Please describe the reason of using the`Lexer::getLocForEndOfToken()` function here.
What would be the difference if `rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1), ")");` was used here?
What would be the difference if  `rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(), ")");` was used here?

Reply:
1) get::LocForEndOfToken() . Computes the source location just past the end of the token at this source location.
2) rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1),")"); give us position right after (cast) and getEndLoc().getLocOffset(1) shifts this position by one to the right. If the variable declared in main has a name of two or more symbol and use rewriter.InsertText(styleCastExpr->getEndLoc().getLocOffset(1),")"); . like main we will get m)ain .
3)rewriter.InsertTextAfterToken(styleCastExpr->getEndLoc(),")"); . If use this we will get the same position and output  but different way.